base/uk.ac.stfc.isis.ibex.ui.scriptgenerator/src/uk/ac/stfc/isis/ibex/ui/scriptgenerator/views/ScriptGeneratorView.java

```diff
@@ -0,0 +1,214 @@
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.layout.RowLayout;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Group;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.wb.swt.ResourceManager;
+import org.eclipse.wb.swt.SWTResourceManager;
+
+import uk.ac.stfc.isis.ibex.scriptgenerator.ActionParameter;
+import uk.ac.stfc.isis.ibex.scriptgenerator.Activator;
+import uk.ac.stfc.isis.ibex.scriptgenerator.ScriptGeneratorSingleton;
+import uk.ac.stfc.isis.ibex.scriptgenerator.table.ActionsTable;
+
+/**
+ * Provides settings to control the script generator.
+ */
+@SuppressWarnings("checkstyle:magicnumber")
+public class ScriptGeneratorView {
+
+	private ScriptGeneratorSingleton scriptGeneratorModel;
+	private Label lblOrder;
+	private DataBindingContext bindingContext;
+	private ActionsTable scriptGeneratorTable;
+	private ActionsViewTable table;
+	private static final Display DISPLAY = Display.getCurrent();
+	
+	private Button createMoveRowButton(Composite parent, String icon, String direction) {
+        Button moveButton =  new Button(parent, SWT.NONE);
+		GridData gdBtnMoveRow = new GridData(SWT.LEFT, SWT.BOTTOM, false, false, 1, 1);
+		gdBtnMoveRow.widthHint = 25;
+		moveButton.setLayoutData(gdBtnMoveRow);
+		moveButton.setImage(ResourceManager.getPluginImage("uk.ac.stfc.isis.ibex.ui", "icons/" + icon));
+        moveButton.setToolTipText("Move selected row " + direction);
+        return moveButton;
+	}
+	
+	/**
+	 * A basic framework to hold the toy interface.
+	 * 
+	 * @param parent the parent composite
+	 */
+	@PostConstruct
+	public void createPartControl(Composite parent) {
+
+		this.scriptGeneratorModel = Activator.getModel();
+		scriptGeneratorTable = this.scriptGeneratorModel.getScriptGeneratorTable();
+		
+		var actionParameters = new ArrayList<ActionParameter>();
+		actionParameters.add(new ActionParameter("column name"));
+		actionParameters.add(new ActionParameter("code"));
+		actionParameters.add(new ActionParameter("third column"));
+		
+		scriptGeneratorModel.setActionParameters(actionParameters);
+		
+		GridData gdQueueContainer = new GridData(SWT.FILL, SWT.FILL, true, false, 1, 1);
+		gdQueueContainer.heightHint = 300;
+		parent.setLayoutData(gdQueueContainer);
+		parent.setLayout(new GridLayout(2, false));
+       
+		table = new ActionsViewTable(parent, SWT.NONE, SWT.SINGLE | SWT.V_SCROLL | SWT.FULL_SELECTION, scriptGeneratorTable);
+		table.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true, 1, 1));
+		
+		//table.addSelectionChangedListener(e -> queueScriptViewModel.setSelectedScript(queuedScriptsViewer.firstSelectedRow()));
+		//addDragAndDrop(queuedScriptsViewer.viewer());
```

Comment:
Is drag and drop in the scope of this ticket? If so do it, if not remove this

Reply:
Alistair has removed it